### ISSUE TITLE
uploadFiles - metadata key just returning dbPath

### ISSUE THREAD

**Issue by raghavendracs** [prescottprue/react-redux-firebase#547]
**Looks like a *bug*?** 

Snapshot of my code is as follows

```
let attachmentUploadPromise = firebase.uploadFiles("/attachments", files, "/allAttachmentsList",options).then((savedObject) => {
     return firebase.database().ref(`/allAttachmentsList/${savedObject[0].key}`).once('value' , snap => {
       this.props.processUploadedAttachments('add', snap.val(),savedObject[0].key)
     })
   }).catch(err => console.log(err))
  }
```

**Current  and Expected Behavior**

Earlier savedObject[0].key used to return the location within dbPath where the metadata was being pushed. But I just realized that with current releases savedObject[0].key just returns the dbPath instead of location at that dbPath where the metadata has been added.

Has the approach to get the key at which metadata is inserted in firebase dbPath different now?

**Comment by prescottprue** [comment]
Hmmm, I can look into it. Which version are you using now?

**Comment by raghavendracs** [comment]
Am using the latest version 2.2.0-alpha.2

**Comment by raghavendracs** [comment]
Any luck - have you had a chance to look please?

**Comment by prescottprue** [comment]
I was attempting to replicate, but I am seeing `key` in the response of both `uploadFiles` and `uploadFile` as expected. Are you using `useFirestoreForStorageMeta`?

While investigating, I did notice that on certain newer version of the Firebase SDK there are undefined values in the metadata returned from storage. It seems unrelated, but going to address as part of the `next` series.

**Comment by prescottprue** [comment]
Fixed as part of `v2.2.0-alpha.3` release. Should work as expected:

```js
firebase.uploadFiles("/attachments", files, "/allAttachmentsList",options)
  .then(([firstFile]) => {
    return firebase.database().ref(`/allAttachmentsList/${firstFile.key}`).ref.once('value')
  })
  .then((snap) => {
     this.props.processUploadedAttachments('add', snap.val(), snap.key)
   })
   .catch(err => console.log(err))
  }
```

Something to note is that the `metadataSnapshot` is included in the response as well so you can simplify your code a little bit:

```js
firebase.uploadFiles("/attachments", files, "/allAttachmentsList",options)
  .then(([firstFile]) => firstFile.metadataSnapshot.ref.once('value'))
  .then((snap) => {
     this.props.processUploadedAttachments('add', snap.val(), snap.key)
   })
   .catch(err => console.log(err))
  }
```

**Comment by raghavendracs** [comment]
Just installed v2.2.0-alpha.3 release. Things are working fine as expected. Thanks.
